File: frontend/src/types/Order.ts

export interface OrderItemRequest {
  productId: number
  quantity: number
}

export interface CompleteOrderRequest {
  items: OrderItemRequest[]
  paymentMethod: 'Cash' | 'Mpesa'
  amountPaid?: number
  transactionId?: string
}

export interface CompleteOrderResponse {
  receiptNumber: string
  total: number
  paymentMethod: string
  amountPaid?: number
  change?: number
  transactionId?: string
  date: string
}

export interface MpesaPaymentRequest {
  customerName: string
  phoneNumber: string
  amount: number
}

export interface MpesaPaymentResponse {
  transactionId: string
  status: string
}

File: backend/Models/Order.cs
namespace backend.Models;

public class Order
{
    public int Id { get; set; }

    public string ReceiptNumber { get; set; } = string.Empty;

    public decimal Total { get; set; }

    public string PaymentMethod { get; set; } = string.Empty;

    public decimal AmountPaid { get; set; }

    public decimal Change { get; set; }

    public string TransactionId { get; set; } = string.Empty;

    public DateTime Date { get; set; }

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}